client/analytics/components/report-table/index.js

```diff
@@ -0,0 +1,138 @@
+
+class ReportTable extends Component {
+	render() {
+		const {
+			getHeadersContent,
+			getRowsContent,
+			getSummary,
+			itemIdField,
+			primaryData,
+			tableData,
+			totalsCountField,
+			// These two props are not used in the render function, but are destructured
+			// so they are not included in the `props` variable.
+			endpoint,
+			tableQuery,
+			...props
+		} = this.props;
+
+		const { items, query } = tableData;
+
+		const isError = tableData.isError || primaryData.isError;
+
+		if ( isError ) {
+			return <ReportError isError />;
+		}
+
+		const isRequesting = tableData.isRequesting || primaryData.isRequesting;
+
+		const headers = getHeadersContent();
+		const orderedItems = orderBy( items, query.orderby, query.order );
+		const ids = orderedItems.map( item => item[ itemIdField ] );
+		const rows = getRowsContent( orderedItems );
+		const totals = get( primaryData, [ 'data', 'totals' ], null );
+		const totalRows = get( totals, [ totalsCountField ], items.length );
+		const summary = getSummary( totals );
+
+		return (
+			<TableCard
+				downloadable
+				headers={ headers }
+				ids={ ids }
+				isLoading={ isRequesting }
+				onQueryChange={ onQueryChange }
+				rows={ rows }
+				rowsPerPage={ query.per_page }
+				summary={ summary }
+				totalRows={ totalRows }
+				{ ...props }
+			/>
+		);
+	}
+}
+
+ReportTable.propTypes = {
+	/**
+	 * The endpoint to use in API calls.
+	 */
+	endpoint: PropTypes.string.isRequired,
+	/**
+	 * A function that returns the headers object to build the table.
+	 */
+	getHeadersContent: PropTypes.func.isRequired,
+	/**
+	 * A function that returns the rows array to build the table.
+	 */
+	getRowsContent: PropTypes.func.isRequired,
+	/**
+	 * A function that returns the summary object to build the table.
+	 */
+	getSummary: PropTypes.func,
+	/**
+	 * The name of the property in the item object which contains the id.
+	 */
+	itemIdField: PropTypes.string.isRequired,
+	/**
+	 * Primary data of that report.
+	 */
+	primaryData: PropTypes.object.isRequired,
+	/**
+	 * Table data of that report.
+	 */
+	tableData: PropTypes.object.isRequired,
+	/**
+	 * Properties to be added to the query sent to the report table endpoint.
+	 */
+	tableQuery: PropTypes.object,
+	/**
+	 * String to display as the title of the table.
+	 */
+	title: PropTypes.string.isRequired,
+	/**
+	 * Name of the property in the primary data totals object which contains the
+	 * total number of items.
+	 */
+	totalsCountField: PropTypes.string,
+};
+
+ReportTable.defaultProps = {
+	getSummary: () => null,
+	tableQuery: {},
+	totalsCountField: '',
+};
+
+export default compose(
+	withSelect( ( select, props ) => {
+		const { endpoint, query, tableQuery } = props;
+		// @TODO allow loading the primary data for the variations table once #926 is fixed.
+		const primaryData =
+			endpoint === 'variations' ? {} : getReportChartData( endpoint, 'primary', query, select );
```

Comment:
Good point. When looking at a single product the chart will display data from `/reports/products/stats` segmented by variations. Would it be possible to construct `primaryData.data.totals` for the table summary numbers without having to query a new endpoint?

Just curious here, even though I think simplicity of a new endpoint may be better.

Reply:
I didn't think about that. Would it be possible to update SwaggerHub or post an example here of how the endpoint response would look like to make sure we are in the same page before implementing it?

Reply:
>Would it be possible to update SwaggerHub or post an example here of how the endpoint response would look like

I think thats a good idea because we'll also run into the same issue on comparisons. I need to think about this some more. Lets tackle this separately.